zipline/utils/calendars/trading_calendar.py

```diff
@@ -635,22 +638,24 @@ def open_and_close_for_session(self, session_label):
         (Timestamp, Timestamp)
             The open and close for the given session.
         """
-        o_and_c = self.schedule.loc[session_label]
+        sched = self.schedule
 
         # `market_open` and `market_close` should be timezone aware, but pandas
         # 0.16.1 does not appear to support this:
         # http://pandas.pydata.org/pandas-docs/stable/whatsnew.html#datetime-with-tz  # noqa
-        return (o_and_c['market_open'].tz_localize('UTC'),
-                o_and_c['market_close'].tz_localize('UTC'))
+        return (
+            sched.at[session_label, 'market_open'].tz_localize('UTC'),
```

Comment:
As long as we're micro-optimizing these functions, it might be worth looking at how this compares to something like:

```
row = self.schedule.index.get_loc(session_label)
return sched.iat[row, 0], sched.iat[row, 1]
```